fhir-path/src/main/java/com/ibm/fhir/path/function/MemberOfFunction.java

```diff
@@ -95,65 +107,88 @@ public int getMaxArity() {
+            if (codeSet != null) {
+                if (codeSet.contains(code)) {
+                    return true;
+                } else {
+                    key = system + "|" + VERSION_UNKNOWN;
+                    codeSet = codeSetMap.get(key);
+                    if (codeSet != null) {
+                        return codeSet.contains(code);
+                    }
+                }
+            }
+        } else if (system != null) {
+            for (String key : codeSetMap.keySet()) {
+                if (key.startsWith(system)) {
+                    return codeSetMap.get(key).contains(code);
+                }
             }
         } else {
-            Compose compose = valueSet.getCompose();
-            if (compose != null) {
-                for (Include include : compose.getInclude()) {
-                    if (!include.getConcept().isEmpty()) {
-                        for (Include.Concept concept : include.getConcept()) {
-                            result.add(concept.getCode().getValue());
-                        }
-                    } else if (include.getSystem() != null){
-                        String system = include.getSystem().getValue();
-                        if (FHIRRegistry.getInstance().hasResource(system)) {
-                            CodeSystem codeSystem = FHIRRegistry.getInstance().getResource(system, CodeSystem.class);
-                            for (CodeSystem.Concept concept : codeSystem.getConcept()) {
-                                result.addAll(getValues(concept));
-                            }
-                        }
-                    }
+            for (Set<String> codeSet : codeSetMap.values()) {
+                if (codeSet.contains(code)) {
+                    return true;
                 }
             }
         }
-        
-        // TODO: add support for exclude
-        
-        return result;
+        return false;
     }
     
-    private Set<String> getValues(Contains contains) {
-        Set<String> values = new LinkedHashSet<>();
-        values.add(contains.getCode().getValue());
-        for (Contains c : contains.getContains()) {
-            values.addAll(getValues(c));
+    private String getSystem(FHIRPathNode node) {
+        if (node == null || !node.isElementNode()) {
+            return null;
         }
-        return values;
-    }
-
-    private Set<String> getValues(CodeSystem.Concept concept) {
-        Set<String> values = new LinkedHashSet<>();
-        values.add(concept.getCode().getValue());
-        for (CodeSystem.Concept c : concept.getConcept()) {
-            values.addAll(getValues(c));
+        for (FHIRPathNode child : node.children()) {
+            if ("system".equals(child.name()) && FHIRPathType.FHIR_URI.equals(node.type())) {
+                return child.asElementNode().element().as(Uri.class).getValue();
+            }
         }
-        return values;
+        return null;
```

Comment:
worth adding method doc to explain when it returns null and what it means?

Reply:
Added some javadoc to the `MemberOfFunction.getSystem` method